src/walk.rs

```diff
@@ -226,63 +231,98 @@ fn spawn_receiver(
                 .unwrap_or_else(|| time::Duration::from_millis(100));
 
             let stdout = io::stdout();
-            let mut stdout = stdout.lock();
+            let stdout = stdout.lock();
+            let mut stdout = io::BufWriter::new(stdout);
 
             let mut num_results = 0;
 
-            for worker_result in rx {
-                match worker_result {
-                    WorkerResult::Entry(value) => {
-                        match mode {
-                            ReceiverMode::Buffering => {
-                                buffer.push(value);
-
-                                // Have we reached the maximum buffer size or maximum buffering time?
-                                if buffer.len() > MAX_BUFFER_LENGTH
-                                    || time::Instant::now() - start > max_buffer_time
-                                {
-                                    // Flush the buffer
-                                    for v in &buffer {
-                                        output::print_entry(
-                                            &mut stdout,
-                                            v,
-                                            &config,
-                                            &wants_to_quit,
-                                        );
+            if config.interactive_terminal {
```

Comment:
If I understand this correctly, this removes the functionality where the results are sorted if the response is fast and short enough if we aren't using an "interactive_terminal", which probably isn't desirable.  I think probably the only difference here should be whether we flush the result after each result.

Reply:
Agreed.  That would simplify the diff quite a bit too, and avoid the duplication between the branches of the `if config.interactive_terminal`.